### PR TITLE
Use default application credential

### PR DESCRIPTION
**Pull request by kroikie**
Update config quickstart to use default application credential which is best practice for server side Firebase authentication.

### CODE REVIEW

File: config/src/main/java/com/google/firebase/samples/config/Configure.java [hunk 1]
```diff
@@ -37,15 +37,20 @@
    * Retrieve a valid access token that can be use to authorize requests to the Remote Config REST
    * API.
    *
+   * This method assumes it is either running in a trusted Google environment like GCP or if running
```
Comment (egilmorez) [inline config/src/main/java/com/google/firebase/samples/config/Configure.java]: Does "it" refer to the server?  Suggest disambiguating.

**Comment by samtstern** [comment]
@kroikie I'm just gonna review the `nodejs` one and then you can assume all comments apply in both places and just merge them both when ready.

**Comment by kroikie** [comment]
@samtstern `nodejs` is ready so this should be as well.

**Comment by kroikie** [comment]
Will only merge when documentation that uses these snippets are ready.

File: config/src/main/java/com/google/firebase/samples/config/Configure.java [hunk 2]
```diff
@@ -35,17 +35,23 @@
...
+   * of the service account credentials file.
    *
    * @return Access token.
    * @throws IOException
    */
   // [START retrieve_access_token]
   private static String getAccessToken() throws IOException {
-    GoogleCredential googleCredential = GoogleCredential
-        .fromStream(new FileInputStream("service-account.json"))
+    GoogleCredential googleCredential = GoogleCredential.getApplicationDefault()
```
Comment (hiranya911) [inline config/src/main/java/com/google/firebase/samples/config/Configure.java]: Should probably use `com.google.auth.oauth2.GoogleCredentials` here.

**Pull request closed without merging**